### PR TITLE
Add test for CSV route checking

### LINKED ISSUE #250: I can output csv files with custom filenames
The csv routes aren’t sufficiently strict about the bit between the final segment of the route and the ‘csv’ ending E.g. both /api/1/access/budget.csv and /api/1/access/budgets.csv work:

 * http://datastore.iatistandard.org/api/1/access/budget.csv
 * http://datastore.iatistandard.org/api/1/access/budgets.csv

In fact, the following custom filename hack also works:

http://datastore.iatistandard.org/api/1/access/budgettt.Hey_look.A_custom_filename.csv

Kind of cool. But probably worth fixing. [I suspect checking the value of format rather than `request.path.endswith` would do the trick.](https://github.com/IATI/IATI-Datastore/blob/30177c55/iati_datastore/iatilib/frontend/api1.py#L226)

### PR DESCRIPTION
**Pull request by andylolz**
Adds tests that fails before 199d9dd3 and pass after.
Also removes default parameter value since it is not used.

Refs #258; Fixes #250.

### CODE REVIEW

File: iati_datastore/iatilib/frontend/api1.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def get(self, format):
 
 
 class DataStoreCSVView(DataStoreView):
-    def get(self, format=".csv"):
+    def get(self, format):
```
Comment (hayfield) [inline iati_datastore/iatilib/frontend/api1.py]: Note: removal of default due to being superfluous https://github.com/IATI/IATI-Datastore/pull/258#issuecomment-292226970

**Comment by andylolz** [comment]
Ooh – #258 is merged now. IIRC, this PR just adds tests to cover the new stuff that was in that PR (as requested by @hayfield here: https://github.com/IATI/IATI-Datastore/pull/258#pullrequestreview-31341473).

**Comment by allthatilk** [comment]
@andylolz if you could merge in latest changes from `master` this one should be ready to test out on our dev Datastore

**Comment by andylolz** [comment]
@allthatilk done

**Pull request merged**